File: exc5_Matriz.rb

mat=[[],[]]

for l in 0..1
    for c in 0..2
        puts "digite o valor para matriz[#{l}] [#{c}]"
        mat [l][c] = gets.chomp.to_i
        puts mat [l][c]
    end
end

puts "====================="
puts "Encontrando intervalo"

sim = 0
nao = 0

for l in 0..1
    for c in 0..2
        intervalo = 0

        if (mat[l][c]>= 5 and mat[l][c] <= 15)
            puts "Notas [#{l}|#{c}]: #{mat[l][c]}"
            puts "Esse numero se encontra no intervalo de 5 e 15"
            sim = sim + 1
        else
            puts "Esse numero nao se encontra no intervalo de 5 e 15"

            nao = nao + 1
        end

        puts "Notas [#{l}|#{c}]: #{mat[l][c]}"

    end
end

puts "======================================================================="
puts "Temos #{sim} numeros que se encontram no intervalo de 5 até 15"
puts "Temos #{nao} numeros que não se encontram no intervalo de 5 até 15"

File: exc5_Matriz.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>
#include <locale.h>

int main()
{
    int mat[2][3], l, c, intervalo, sim, n;
    
    setlocale(LC_ALL, "Portuguese");
    
    for(l=0; l<=1; l++){
        for(c=0; c<=2; c++){
            scanf("%d", &mat[l][c]);
        }
    }
    
    printf("=====================");
    printf("\nEncontrando intervalo");
    
    sim = 0;
	n = 0;
	
	for(l=0; l<=1; l++){
        for(c=0; c<=2; c++){
            
            intervalo = 0;
            
            if(mat[l][c] >= 5 && mat[l][c] <= 15){
                printf("\nNotas: %d ", mat[l][c]);
                printf("\nEsse numero se encontra no intervalo de 5 e 15");
                
                sim = sim + 1;
            }
            
            else{
                printf("\nEsse numero nao se encontra no intervalo de 5 e 15");
                
                n = n + 1;
            }
            
            printf("\nNotas: %d ", mat[l][c]);
            
        }
    }
    
    printf("=======================================================================");
    printf("\n Temos %d numeros que se encontram no intervalo de 5 até 15", sim);
    printf("\n Temos %d numeros que não se encontram no intervalo de 5 até 15", n);
    
    return 0;
}
